### PR TITLE
Add RPC API tests

### PR DESCRIPTION
**Pull request by evilrobot-01**
- added rpc api tests (simple mock of runtime api)
- updated workflow to install protobuf-compiler due to missing protoc error after addition of tests
- used actions-rs/toolchain for Rust installation, related to resolving above error and seems to be commonly used in paritytech repos (https://github.com/search?q=org%3Aparitytech+actions-rs&type=code)
- added flag to skip wasm build, eliminating need for nightly

### CODE REVIEW

File: rpc/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+
+fn assert_overflow(error: Error) {
+    assert!(matches!(error, Error::Call(e) if matches!(&e, CallError::Custom(e)
+        if e.code() == OVERFLOW && e.message() == OVERFLOW_MESSAGE && e.data().is_none())));
+}
+
+fn assert_unexpected(error: Error) {
+    assert!(matches!(error, Error::Call(e) if matches!(&e, CallError::Custom(e)
+        if e.code() == RUNTIME_ERROR && e.message() == RUNTIME_ERROR_MESSAGE && e.data().is_some())));
+}
```
Comment (Wiezzel) [inline rpc/src/tests.rs:38]: These 4 functions could be refactored into a single one, taking `code` and `message` as params.

**Review by Wiezzel — COMMENTED** [review]
One small comment, overall 👍🏻

**Comment by evilrobot-01** [comment]
Good idea, will get implemented later today. 🙂
________________________________

**Comment by evilrobot-01** [comment]
Ps - if you're happy to merge the other PR now, I can update this branch with rebase.

File: rpc/src/tests.rs (hunk 1 above)
Comment (evilrobot-01) [reply rpc/src/tests.rs]: Done

**Comment by Wiezzel** [comment]
@evilrobot-01 The other one is merged, please rebase

**Comment by evilrobot-01** [comment]
Just used the GitHub UI to sync fork via 'update branch' and its unfortunately created an additional merge commit, which isn't all that clean. Are you possibly fine with accepting as is to get this finished?

**Comment by Wiezzel** [comment]
> Just used the GitHub UI to sync fork via 'update branch' and its unfortunately created an additional merge commit, which isn't all that clean. Are you possibly fine with accepting as is to get this finished?

Not a problem for me.

**Comment by evilrobot-01** [comment]
Thanks, good to go when you have a moment. 🙂

**Pull request merged**